lib/aruco_detector.hpp

```diff
@@ -0,0 +1,55 @@
+#include <opencv2/opencv.hpp>
+#include <opencv2/aruco.hpp>
+#include <memory>
+#include "classification_model.hpp"
+
+class Aruco_Detector : public ClassificationModel::dClassification {
+public:
+    Aruco_Detector(cv::VideoCapture& videoCapture, int dictionaryType) : inputVideo(videoCapture), dictionaryType(dictionaryType) {}
+
+    ClassificationModel::Detection classify(cv::Mat& image, float threshold) override {
+        if (detectMarkers(image)) {
+            return ClassificationModel::Detection::ARUCO;
+        } else {
+            return ClassificationModel::Detection::NONE;
+        }
+    }
+
+    cv::Rect get_bounding_box() override {
+        return cv::Rect();
+    }
+
+private:
+    int dictionaryType;
+    ArUcoDetectionResult detectionResult;
+
+    void detectMarkers(cv::Mat& image) {
+        //Camera/Marker Parameters
+        cv::Mat cameraMatrix = (cv::Mat_<double>(3, 3) << 216.357407, 0, 214.624283,
+                0, 216.357407, 115.447563,
+                0, 0, 1);
+        cv::Mat distCoeffs = cv::Mat::zeros(5, 1, CV_64F);
+        cv::Ptr<cv::aruco::Dictionary> dictionary = cv::aruco::getPredefinedDictionary(dictionaryType);
+
+        if (image.empty()) {
+            std::cerr << "Error: Input image is empty." << std::endl;
+            return;
+        }
+
+        std::vector<int> ids;
+        std::vector<std::vector<cv::Point2f>> corners;
+        cv::aruco::detectMarkers(image, dictionary, corners, ids);
+
+        // If at least one marker is detected
+        if (!ids.empty()) {
+            std::vector<cv::Vec3d> rvecs, tvecs;
+            cv::aruco::estimatePoseSingleMarkers(corners, 0.15, cameraMatrix, distCoeffs, rvecs, tvecs);
```

Comment:
Take 0.15 as an argument or a model parameter

Reply:
If there are too many parameters, maybe make a struct for them, something like: https://github.com/kknives/michi/blob/fd030796cc2bcc88a81b5c568af13e30245abc91/lib/yolov8_arrow.hpp#L29-L46

Reply:
i did not understand this

Reply:
No, but it can be changed if the border for the Aruco tags changes.

Reply:
But not during runtime, so maybe they can be added into the Params struct or taken as a parameter in the constructor